clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -185,16 +185,50 @@ private EcsServerGroup generateServerGroup(AmazonCredentials.AWSRegion awsRegion
                                              ServiceMetadata metadata,
                                              Set<Instance> instances,
                                              ServerGroup.Capacity capacity) {
+    ServerGroup.InstanceCounts instanceCounts = generateInstanceCount(instances);
+
     return new EcsServerGroup()
       .setName(constructServerGroupName(metadata))
       .setCloudProvider(EcsCloudProvider.ID)
       .setType(EcsCloudProvider.ID)
       .setRegion(awsRegion.getName())
       .setInstances(instances)
       .setCapacity(capacity)
+      .setInstanceCounts(instanceCounts)
       ;
   }
 
+  private ServerGroup.InstanceCounts generateInstanceCount(Set<Instance> instances) {
+    ServerGroup.InstanceCounts instanceCounts = new ServerGroup.InstanceCounts();
+    for (Instance instance: instances) {
+      switch (instance.getHealthState()) {
+        case Up:
+          instanceCounts.setUp(instanceCounts.getUp() + 1);
+          break;
+        case Down:
+          instanceCounts.setDown(instanceCounts.getDown() + 1);
+          break;
+        case Failed:
+          instanceCounts.setDown(instanceCounts.getDown() + 1);
+          break;
+        case Starting:
+          instanceCounts.setOutOfService(instanceCounts.getOutOfService() + 1);
+          break;
+        case Unknown:
```

Comment:
Does this case statement cover all the possible values of instance.getHealthState()? It might be better to move the unknown case to the default and catch all. Or add a default that raises an error if something different is found.

Reply:
`Unknown` is not a java convention, but rather a value in the HealthState enum:

```
enum HealthState {
  Up, Down, Unknown, Starting, OutOfService, Succeeded, Failed
}
```

All enum values are covered by the switch statement - the default catch-all case would never get hit here.  I sometimes like making the default case throw an error, just so that i can fail quick.  Helps debugging tremendously.  That being said, the HealthState enum is very unlikely to change